Project-Descriptions-and-Plans/LLVM-Toolchain/COREV_LLVM_Plan_Approved_2023.md

```diff
@@ -0,0 +1,173 @@
+
+Everyone who wants to develop software for CORE-V family of cores and devices implementing the CORE-V extensions will need a compiler.  They have a choice to use the CORE-V GCC compiler tool chain or the CORE-V Clang/LLVM tool chain (this project).
+
+
+## Project license model
+
+See License Scheme above.
+
+## Description of initial code contribution, if required
+
+embecosm - V1
+
+## Repository Structure
+
+The full code will initially be stored in the following github repository:
+https://github.com/openhwgroup/corev-llvm-project
+
+The upstream code which is the eventual target for this project is in the following repository:
+https://github.com/llvm/llvm-project
+
+## Project distribution model
+
+OpenHW GitHub repository
+
+## Project plan
+
+### 2023 workplan
+
+| Related TG         | Milestone                                                                                                                      | Target        | Contributor |
+| ------------------ | ------------------------------------------------------------------------------------------------------------------------------ | ------------- | ----------- |
+| Assembler          | re-encode the existing CV32E40Pv1 instructions                                                                                 | 2023-02-28    | ISCAS       |
+| Assembler          | SIMD instructions                                                                                                              | 2023-03-30    | ISCAS       |
+| Assembler          | Bitmanip                                                                                                                       | 2023-04-30    | ISCAS       |
+| Compiler options   | CV32E40Pv2 has its own compilation options, support it. (eg: xcvhwlp,xcvsimd…)                                                 | 2023-05-30    | ISCAS       |
+| Codegen            | SIMD builtins                                                                                                                  | 2023-07-30    | ISCAS       |
+| Codegen            | Bitmanip builtins                                                                                                              | 2023-09-30    | ISCAS       |
+| Codegen            | General ALU                                                                                                                    | 2023-10-30    | ISCAS       |
+| Codegen            | General load/store                                                                                                             | 2023-11-30    | ISCAS       |
+| Hardware loop      | CV32E40Pv1 has already implemented version 1, and new features will be added to it later                                       | 2023-12-30    | ISCAS       |
+
+### Project tracking and meetings
+
+The progress towards 2023 milestones will be tracked in progress meetings. Slides will be updated during the meeting and posted on LLVM Tools Mattermost channel.
```

Comment:
Are you going to use a Kanban Board to track the progress? That seems helpful to keep everyone in the loop

Reply:
I think we can use the Kanban board, but how exactly to operate, I am not very clear.

Reply:
ok, I found the kanban channel.

Reply:
GitJHub project boards are a suitable technology if desired.  The first step is the list of tasks and dates, which we have below.